src/Data/Text/Encoding.hs

```diff
@@ -208,53 +214,120 @@ decodeAsciiPrefix bs = if B.null bs
+#ifdef SIMDUTF
+      isValidBS off bLen bs = withBS bs $ \ fp _ -> unsafeDupablePerformIO $
+        unsafeWithForeignPtr fp $ \ptr -> (/= 0) <$> c_is_valid_utf8 (ptr `plusPtr` off) (fromIntegral bLen)
+#else
+#if MIN_VERSION_bytestring(0,11,2)
+      isValidBS off bLen = B.isValidUtf8 . B.take bLen . B.drop off
+#else
+      isValidBS off bLen bs = start off
+        where
+          start ix
+            | ix >= off + bLen = True
+            | otherwise = case utf8DetectStart (B.unsafeIndex bs ix) of
+              Accept -> start (ix + 1)
+              Reject -> False
+              Incomplete st -> step (ix + 1) st
+          step ix st
+            | ix >= off + bLen = False
+            | otherwise = case utf8DetectContinue (B.unsafeIndex bs ix) st of
+              Accept -> start (ix + 1)
+              Reject -> False
+              Incomplete st' -> step (ix + 1) st'
+#endif
+#endif
+      bsToText bs count dst dstOff =
+        when (count > 0) . withBS bs $ \ fp _ ->
+          unsafeIOToST . unsafeWithForeignPtr fp $ \ src ->
+            unsafeSTToIO $ A.copyFromPointer dst dstOff src count
+      chunksToText utf8Len
+        | utf8Len > 0 = runST $ do
+          dst <- A.new utf8Len
+          let crossesBsBoundary = utf8Len > len1
+          bsToText bs1 (if crossesBsBoundary then len1 else utf8Len) dst 0
+          when crossesBsBoundary $ bsToText bs2 (utf8Len - len1) dst len1
+          arr <- A.unsafeFreeze dst
+          pure $ Text arr 0 utf8Len
+        | otherwise = empty
+      decodeResult isErr off =
+        let off' = if isErr then off + 1 else off in
+        DecodeResult
+          (chunksToText off)
+          (if isErr then Just $ index off else Nothing)
+          (if off' < len1
+            then B.drop off' bs1 `B.append` bs2
+            else B.drop (off' - len1) bs2)
+          off'
+      countValidUtf8 i _ Reject = decodeResult True i
+      countValidUtf8 i i' (Incomplete a)
+        | i' < len = countValidUtf8 i (i' + 1) $ utf8DetectContinue (index i') a
+        | otherwise = decodeResult False i
+      countValidUtf8 _ i' _
+        | i' < len = countValidUtf8 i' (i' + 1) . utf8DetectStart $ index i'
+        | otherwise = decodeResult False i'
+      wrapUpBs1 off = countValidUtf8 off off Accept
+      wrapUpBs2 off =
+        wrapUpBs1 $ if bs2Utf8Boundary > off && isValidBS (off - len1) (bs2Utf8Boundary - off) bs2
+          then bs2Utf8Boundary
+          else off
+  in
+  if bs1Utf8Boundary > 0
+  then
+    if isValidBS 0 bs1Utf8Boundary bs1
+    then
+      if bs1Utf8Boundary < len1
+      then
+        let checkCodePointAccrossBoundary Reject _ = decodeResult True bs1Utf8Boundary
+            checkCodePointAccrossBoundary (Incomplete a) off = if off < len
+              then checkCodePointAccrossBoundary (utf8DetectContinue (index off) a) $ off + 1
+              else decodeResult False bs1Utf8Boundary
+            checkCodePointAccrossBoundary Accept off = wrapUpBs2 off
```

Comment:
It's at the end of the first chunk and the beginning of the second. It validates a code point that crosses over the chunks. So this can happen, and when it does, the offset is one to three words into the second chunk.